sources/packages/backend/apps/queue-consumers/src/processors/schedulers/institution-integration/ece-response/_tests_/ece-response-integration.scheduler.e2e-spec.ts

```diff
@@ -0,0 +1,574 @@
+      ];
+      expectedResult.warnings = [
+        `Disbursement ${disbursement.id}, record skipped due to reason: Enrolment for the given application not found.`,
+      ];
+      expect(processResult).toStrictEqual([expectedResult]);
+      // Expect the delete method to be called.
+      expect(sftpClientMock.delete).toHaveBeenCalled();
+    });
+
+    it("Should stop processing the ECE response file when the header record is not valid.", async () => {
+      // Arrange
+      const confirmEnrolmentInstitutionCode = "FAIL";
+      const confirmEnrolmentResponseFile = path.join(
+        process.env.INSTITUTION_RESPONSE_FOLDER,
+        confirmEnrolmentInstitutionCode,
+        ECE_RESPONSE_FILE_NAME,
+      );
+      // Mock the implementation to return the mocked institution code.
+      institutionLocationService.getAllIntegrationEnabledInstitutionCodes =
+        async () => {
+          return [confirmEnrolmentInstitutionCode];
+        };
+
+      // Queued job.
+      const job = createMock<Job<void>>();
+
+      // Modify the data in mock file to have invalid header.
+      mockDownloadFiles(
+        sftpClientMock,
+        [ECE_RESPONSE_FILE_NAME],
+        (fileContent: string) => {
+          return fileContent.replace("1AJAA", "2AJAA");
+        },
+      );
+
+      // Act
+      const processResult = await processor.processECEResponse(job);
+
+      // Assert
+      const expectedResult: ProcessSummaryResult = new ProcessSummaryResult();
+      expectedResult.summary = [
+        `Starting download of file ${confirmEnrolmentResponseFile}.`,
+        `The file ${confirmEnrolmentResponseFile} has been deleted after processing.`,
+      ];
+      expectedResult.errors = [
+        `Error processing the file ${confirmEnrolmentResponseFile}. Error: The ECE response file has an invalid record type on header: 2`,
+      ];
+      expect(processResult).toStrictEqual([expectedResult]);
+      // Expect the delete method to be called.
+      expect(sftpClientMock.delete).toHaveBeenCalled();
+    });
+
+    it("Should stop processing the ECE response file when the detail record is not valid.", async () => {
+      // Arrange
+      const confirmEnrolmentInstitutionCode = "FAIL";
+      const confirmEnrolmentResponseFile = path.join(
+        process.env.INSTITUTION_RESPONSE_FOLDER,
+        confirmEnrolmentInstitutionCode,
+        ECE_RESPONSE_FILE_NAME,
+      );
+      // Mock the implementation to return the mocked institution code.
+      institutionLocationService.getAllIntegrationEnabledInstitutionCodes =
+        async () => {
+          return [confirmEnrolmentInstitutionCode];
+        };
+
+      // Queued job.
+      const job = createMock<Job<void>>();
+
+      // Modify the data in mock file to have invalid detail.
+      mockDownloadFiles(
+        sftpClientMock,
+        [ECE_RESPONSE_FILE_NAME],
+        (fileContent: string) => {
+          return fileContent.replace("2AJBH", "3AJBH");
```

Comment:
The `replace` will act on the first occurrence only. The code can be `fileContent.replace("2", "3");`

Reply:
I assume the file content is the complete string of file. The first occurrence of 2 is already in line 1.